src/coreclr/pal/inc/pal.h

```diff
@@ -3618,6 +3618,22 @@ Define_InterlockMethod(
     ((PVOID)(UINT_PTR)InterlockedCompareExchange((PLONG)(UINT_PTR)(Destination), (LONG)(UINT_PTR)(ExChange), (LONG)(UINT_PTR)(Comperand)))
 #endif
 
+#if defined(HOST_64BIT) && defined(FEATURE_CACHED_INTERFACE_DISPATCH)
+FORCEINLINE uint8_t _InterlockedCompareExchange128(int64_t volatile *pDst, int64_t iValueHigh, int64_t iValueLow, int64_t *pComparandAndResult)
+{
+    __int128_t iComparand = ((__int128_t)pComparandAndResult[1] << 64) + (uint64_t)pComparandAndResult[0];
+    // TODO-LOONGARCH64: for LoongArch64, it supports 128bits atomic from 3A6000-CPU which is ISA1.1's version.
+    // The LA64's compiler will translate the `__sync_val_compare_and_swap` into calling the libatomic's library interface to emulate
+    // the 128-bit CAS by mutex_lock if the target processor doesn't support the ISA1.1.
+    // But this emulation by libatomic doesn't satisfy requirements here which it must update two adjacent pointers atomically.
+    // this is being discussed in https://github.com/dotnet/runtime/issues/109276.
```

Comment:
Note that per this comment, it will drop the LA64 ISA 1.0 support from coreclr. Currently 1.1 requirement is exclusive to nativeaot.